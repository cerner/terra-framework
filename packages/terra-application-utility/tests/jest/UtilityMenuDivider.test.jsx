import React from 'react';
import Divider from '../../src/utility/_UtilityMenuDivider';

describe('Divider', () => {
  const defaultRender = <Divider />;

  it('should render a default utility menu divider', () => {
    const wrapper = shallow(defaultRender);
    expect(wrapper).toMatchSnapshot();
  });

  it('it should pass in a custom prop', () => {
    const wrapper = shallow(<Divider id="testDivider" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('correctly applies the theme context className', () => {
    jest.spyOn(React, 'useContext')
      .mockReturnValue({
        className: 'orion-fusion-theme',
      });
    const wrapper = mount(defaultRender);
    expect(wrapper).toMatchSnapshot();
  });
});
