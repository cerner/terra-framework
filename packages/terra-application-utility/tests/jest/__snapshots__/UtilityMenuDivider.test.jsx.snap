// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider it should pass in a custom prop 1`] = `
<div
  className="divider is-bottom"
  id="testDivider"
  role="separator"
/>
`;

exports[`Divider should render a default utility menu divider 1`] = `
<div
  className="divider is-bottom"
  role="separator"
/>
`;
